### PR TITLE
feat(diffusers/models/unets): add unet1d & StableCascade

### PR DESCRIPTION
**Pull request by Cui-yshoho**
# What does this PR do?

Fixes # (issue)

Adds # (feature)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/mindspore-lab/mindone/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the documentation with your changes? E.g. record bug fixes or new features in `What's New`. Here are the
      [documentation guidelines](https://github.com/mindspore-lab/mindcv/wiki/%E6%96%87%E6%A1%A3%E7%BC%96%E5%86%99%E8%A7%84%E8%8C%83)
- [ ] Did you build and run the code without any errors?
- [ ] Did you report the running environment (NPU type/MS version) and performance in the doc? (better record it for data loading, model inference, or training tasks)
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@xxx

### CODE REVIEW

**Comment by Cui-yshoho** [comment]
Supplement: https://gist.github.com/Cui-yshoho/7ff86a76323c37f1c197d7fef67702ec

**Review by CaitinZhao — APPROVED** [review]

File: mindone/diffusers/utils/mindspore_utils.py [hunk 1]
```diff
@@ -57,3 +57,54 @@ def randn_tensor(
         latents = randn(shape, generator=generator, dtype=dtype)
 
     return latents
+
+
+def ms_conv_transpose1d(input, weight, bias=None, stride=1, padding=0, output_padding=0, groups=1, dilation=1):
```
Comment (geniuspatrick) [inline mindone/diffusers/utils/mindspore_utils.py:62]: 这个为什么单独实现一个？

File: mindone/diffusers/models/unets/unet_stable_cascade.py [hunk 2]
```diff
@@ -0,0 +1,627 @@
...
+            elif block_type == "SDCascadeTimestepBlock":
+                return SDCascadeTimestepBlock(
+                    in_channels, timestep_ratio_embedding_dim, conds=timestep_conditioning_type
+                )
+            else:
+                raise ValueError(f"Block type {block_type} not supported")
+
+        # BLOCKS
+        # -- down blocks
+        down_blocks = nn.CellList()
```
[the remark this answers is not in the document]
Comment (Cui-yshoho) [reply mindone/diffusers/models/unets/unet_stable_cascade.py]: 已修改

File: mindone/diffusers/models/unets/unet_1d.py [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+            fc_dim=block_out_channels[-1] // 4,
+        )
+
+        self.use_timestep_embedding = self.config.use_timestep_embedding
+
+    def construct(
+        self,
+        sample: ms.Tensor,
+        timestep: Union[ms.Tensor, float, int],
+        return_dict: bool = True,
```
[the remark this answers is not in the document]
Comment (Cui-yshoho) [reply mindone/diffusers/models/unets/unet_1d.py]: 已修改

**Review by geniuspatrick — APPROVED** [review]

**Pull request merged**